src/benchInsert.c

```diff
@@ -1710,6 +1711,151 @@ void loadChildTableInfo(threadInfo* pThreadInfo) {
+    if (NULL == pThreadInfo->conn->stmt2) {
+        errorPrint("reCreateConn taos_stmt2_init() failed, reason: %s\n", taos_errstr(NULL));
+        return -1;
+    } else {
+        succPrint("%s", "reCreateConn second taos_stmt2_init() success.\n");
+    }
+
+    // select db 
+    if (taos_select_db(pThreadInfo->conn->taos, pThreadInfo->dbInfo->dbName)) {
+        errorPrint("second taos select database(%s) failed\n", pThreadInfo->dbInfo->dbName);
+        return -1;
+    }
+
+    return 0;
+}
+
+// reinit
+int32_t reinitStmt2(threadInfo * pThreadInfo, int32_t w) {
+    // re-create connection
+    int32_t code = reCreateConn(pThreadInfo);
+    if (code != 0) {
+        return code;
+    }
+
+    // prepare
+    code = prepareStmt2(pThreadInfo->conn->stmt2, pThreadInfo->stbInfo, NULL, w);
+    if (code != 0) {
+        return code;
+    }
+
+    return code;
+}
+
+int32_t submitStmt2Impl(threadInfo * pThreadInfo, TAOS_STMT2_BINDV *bindv, int64_t *delay1, int64_t *delay3,
+                    int64_t* startTs, int64_t* endTs, uint32_t* generated) {
+    // call bind
+    int64_t start = toolsGetTimestampUs();
+    int32_t code = taos_stmt2_bind_param(pThreadInfo->conn->stmt2, bindv, -1);
+    if (code != 0) {
+        errorPrint("taos_stmt2_bind_param failed, reason: %s\n", taos_stmt2_error(pThreadInfo->conn->stmt2));
+        return code;
+    }
+    debugPrint("interlace taos_stmt2_bind_param() ok.  bindv->count=%d \n", bindv->count);
+    *delay1 += toolsGetTimestampUs() - start;
+
+    // execute
+    *startTs = toolsGetTimestampUs();
+    code = execInsert(pThreadInfo, *generated, delay3);
+    *endTs = toolsGetTimestampUs();
+    return code;
+}
+
+int32_t submitStmt2(threadInfo * pThreadInfo, TAOS_STMT2_BINDV *bindv, int64_t *delay1, int64_t *delay3,
+                    int64_t* startTs, int64_t* endTs, uint32_t* generated, int32_t w) {
+    // calc loop
+    int32_t loop = 1;
+    SSuperTable* stbInfo = pThreadInfo->stbInfo;
+    if(stbInfo->continueIfFail == YES_IF_FAILED) {          
+        if(stbInfo->keep_trying > 1) {
+            loop = stbInfo->keep_trying;
+        } else {
+            loop = 3; // default
+        }
+    }
+
+    // submit stmt2
+    int i = 0;
+    while (1) {
+        int32_t code = submitStmt2Impl(pThreadInfo, bindv, delay1, delay3, startTs, endTs, generated);
+        if ( code == 0) {
+            // success
+            break;
+        } else {
+            // failed
+            if (loop == 0) {
+                // failed for ever
+                errorPrint("finally faild execute submitStmt2() after retry %d \n", i);
+                return -1;
+            }
+            loop --;
+
+            // wait a memont for trying
+            toolsMsleep(stbInfo->trying_interval);
+            // reinit
+            infoPrint("stmt2 start retry submit i=%d  after sleep %d ms...\n", i++, stbInfo->trying_interval);
+            code = reinitStmt2(pThreadInfo, w);

```

Comment:
if reinit failed , do not call submitStmt2Impl again

Reply:
done